app/controllers/api/registrations_controller.rb

```diff
@@ -3,7 +3,7 @@ class API::RegistrationsController < Devise::RegistrationsController
 
   def create
     build_resource(sign_up_params)
-    if resource.save
+    if can_create? && resource.save
```

Comment:
We don't really want to render 422 if can_create? is false. I think a before_action is the way to go here.

```
actor.ability.authorize! :create, resource
```

Then `can_create?` goes into a :create, User ability block, which is the most natural place for it

Reply:
knew you'd have the answer for that. Thanks